internal/transformations/static_analyzer.go

```diff
@@ -164,6 +164,9 @@ func analyzeVisit(a ast.Node, inObject bool, vars ast.IdentifierSet) error {
 	return s.err
 }
 
-func analyze(node ast.Node) error {
+// Analyze checks variable references (these could be checked statically in Jsonnet).
+// It enriches ast with additional information about free variables in every node,
+// so it is necessary to always run it before executing AST.
+func Analyze(node ast.Node) error {
```

Comment:
Does anything call on this function from outside the package? Do we need to export it?

Reply:
No, I unexported it. 

Originally Analyze and Desugar were supposed to be high-level operations here, and SnippetToAST was just a helper function in my original idea for this package. But since that's the only part that is actually used, it makes sense to unexport these two.